kolibri/plugins/learn/assets/src/views/index.vue

```diff
@@ -39,26 +39,28 @@
 
     <div v-if="!isSearchPage">
       <tabs>
-        <tab-link
-          type="icon-and-title"
-          :title="$tr('recommended')"
-          icon="forum"
-          :link="recommendedLink"
-        />
-        <tab-link
-          type="icon-and-title"
-          :title="$tr('topics')"
-          icon="folder"
-          :link="topicsLink"
-        />
-        <tab-link
-          name="exam-link"
-          v-if="isUserLoggedIn && userHasMemberships"
-          type="icon-and-title"
-          :title="$tr('exams')"
-          icon="assignments"
-          :link="examsLink"
-        />
+        <template scope="tab">
+          <component
+            :is="tab.link"
```

Comment:
> I'm curious about why we're not just passing in an array or something, though

I mainly wanted to avoid having complex data structures (e.g. arrays of objects) as props because they're harder to validate and document.

> it just seems like we're writing a bunch of repetitious code for no particular reason

I'm admittedly not the sharpest tool in the shed, but I find scoped slots to be rather confusing. Would they actually reduce code repetition?